src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -229,50 +223,32 @@ MethodSymbol setMethod(MethodSymbol newMethod) {
      *  @param sym        The deprecated symbol.
      */
     void warnDeprecated(DiagnosticPosition pos, Symbol sym) {
-        LintWarning warningKey = null;
-        if (sym.isDeprecatedForRemoval()) {
-            if (!lint.isSuppressed(LintCategory.REMOVAL)) {
-                if (sym.kind == MDL) {
-                    warningKey = LintWarnings.HasBeenDeprecatedForRemovalModule(sym);
-                } else {
-                    warningKey = LintWarnings.HasBeenDeprecatedForRemoval(sym, sym.location());
-                }
-            }
-        } else if (!lint.isSuppressed(LintCategory.DEPRECATION)) {
-            if (sym.kind == MDL) {
-                warningKey = LintWarnings.HasBeenDeprecatedModule(sym);
-            } else {
-                warningKey = LintWarnings.HasBeenDeprecated(sym, sym.location());
-            }
-        }
-        if (warningKey != null)
-            log.warning(pos, warningKey);
+        Assert.check(!importSuppression);
+        LintWarning warningKey = sym.isDeprecatedForRemoval() ?
```

Comment:
Another big win is the absence of all "is enabled?" checks in the new code. Meaning less potential for mistakes.